keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,525 @@
+     * This operation can only be enabled on a soft-delete enabled vault. This operation requires the secrets/purge permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return A {@link Mono} containing a {@link RestVoidResponse}.
+     * @throws com.azure.common.http.rest.RestException when a deleted secret with {@code name} doesn't exist in the key vault.
+     */
+    public Mono<RestVoidResponse> purgeDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.purgeDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+    /**
+     * Recovers the deleted secret in the key vault to its latest version.
+     * This operation can only be performed on a soft-delete enabled vault.
+     * This operation requires the secrets/recover permission.
+     *
+     * @param name The name of the deleted secret to be recovered.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return A {@link Mono} containing a {@link RestResponse} whose {@link RestResponse#body()} contains the recovered {@link Secret}.
+     * @throws com.azure.common.http.rest.RestException when a deleted secret with {@code name} doesn't exist in the key vault.
+     */
+    public Mono<RestResponse<Secret>> recoverDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.recoverDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+    /**
+     * Requests a backup of the specified secret be downloaded to the client. All versions of the secret will be downloaded.
+     * This operation requires the secrets/backup permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return A {@link Mono} containing a {@link RestResponse} whose {@link RestResponse#body()} contains the backed up secret blob.
+     * @throws com.azure.common.http.rest.RestException when a secret with {@code name} doesn't exist in the key vault.
+     */
+    public Mono<RestResponse<byte[]>> backupSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.backupSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE)
+                .flatMap(new Function<RestResponse<SecretBackup>, Mono<? extends RestResponse<byte[]>>>() {
+                    @Override
+                   public Mono<? extends RestResponse<byte[]>> apply(RestResponse<SecretBackup> base64URLRestResponse) {
+                        return Mono.just(new SimpleRestResponse<byte[]>(base64URLRestResponse.request(),
+                            base64URLRestResponse.statusCode(), base64URLRestResponse.headers(), base64URLRestResponse.body().value()));
+                    }
+                });
```

Comment:
You should use a lambda expression here rather than an anonymous inner class - it'll get rid of a lot of the noise here (lines 370, 371, and 372 would basically disappear).

Reply:
True. Added lambda expression.